### ISSUE TITLE
Configuration option for eager host <-> device data transfer

### ISSUE THREAD

**Issue by kynan** [OP2/PyOP2#399]
Add a configuration option to trigger data transfer eagerly instead of marking data as out of date. This may be advantageous in scenarios where the user knows data is going to be accessed by starting the transfer earlier and potentially overlapping with computation on the host.

**Comment by wence-** [comment]
No longer relevant after #507.
